influent-java/src/main/java/influent/forward/ForwardClient.java

```diff
@@ -0,0 +1,101 @@
+package influent.forward;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Client ip/network authentication & per_host shared key
+ */
+public class ForwardClient {
+  private String host = null;
+  private String network = null;
+  private String sharedKey = null;
+  private List<String> usernames = new ArrayList<>();
+
+  public ForwardClient(String host,
+                       String network,
+                       String sharedKey,
+                       List<String> usernames) {
+    this.host = host;
+    this.network = network;
+    this.sharedKey = sharedKey;
+    this.usernames = usernames;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public String getNetwork() {
+    return network;
+  }
+
+  public String getSharedKey() {
+    return sharedKey;
+  }
+
+  public List<String> getUsernames() {
+    return usernames;
+  }
+
+  public static class Builder {
+    private String host = null;
+    private String network = null;
```

Comment:
I want to validate these arguments and create `InetAddress` or `InetNetwork` instances early.

Reply:
Do you mean like following?

```java
private InetAddress host = null;
private InetNetwork network = null;
// ...
public static Builder ofHost(String host) {
  Builder builder = new Builder();
  builder.host = InetAddress.getByName(host);
  return builder;
}
public static Builder ofNetwork(String network) {
  Builder builder = new Builder();
  builder.network = InetNetwork.getBySpec(network);
  return builder;
}
```

Reply:
Yes.
However, errors are detected before starting a server since host/network formats are validated on creating ForwardSecurity.
So there is no problem with the current implementation.